packages/svelte2tsx/test/sourcemaps/process.ts

```diff
@@ -0,0 +1,498 @@
+    const backward = generated.text.lastIndexOf(generated_subset, fromLineCharToOffset(cue));
+    const target = forward === backward || forward === -1 ? backward : forward;
+    return target;
+
+    function tryPickMatch(matches: MappedPosition[]) {
+        const exact = matches.filter((match) =>
+            match.generated.line.source.text
+                .slice(fromLineCharToOffset(match.generated))
+                .startsWith(generated_subset)
+        );
+        if (exact.length === 1) return exact[0];
+        const l = exact.length;
+        const m = matches.length;
+        const m_of_them = l === m ? 'all of them' : l === 0 ? 'none' : `${l} out of ${m}`;
+        throw new Error(
+            `Could not find TestRange: Generated text includes ` +
+                `${m} characters mapping back to origin's ${toString(matches[0].original)} and ` +
+                `${m_of_them} start with the tested text "${generated_subset}"` +
+                `\n Matching : ${matches.map((match) => toString(match.generated)).join(',\n')}`
+        );
+    }
+
+    function toString(pos: Position) {
+        return `[${pos.line.index + 1}:${pos.character + 1}]`;
+    }
+}
+
+export function validate_edit_file(text_with_ranges: string) {
+    for (const raw of parse_edit_ranges(text_with_ranges, false)) {
+        if (raw.start === -1 || raw.end === -1) {
+            const missing_start = raw.start === -1;
+            const end = missing_start ? 'start' : 'end';
+            const start = missing_start ? 'end' : 'start';
+            const index = missing_start ? raw.end : raw.start;
+            const { line, character } = new SourceText(text_with_ranges).toLineChar(index);
+            throw new Error(
+                `Line ${line.index + 1} ${start}s a range that has no corresponding ${end}\n\n` +
+                    `\t${line.print()}\n` +
+                    `\t${' '.repeat(character) + '^^^'}\n`
+            );
+        }
+    }
+}
+
+export function validate_test_file(test: string) {
+    raw.fromTestFile(test);
+}
+
+export function is_edit_changed(edit_file: string) {
+    try {
+        const { content, hash: prev } = raw.fromEditFile(edit_file);
+        return hash(content) !== prev;
+    } catch {
+        return true;
+    }
+}
+
+export function is_test_from_same_input(test: string, input: string) {
+    return hash(input) === raw.fromTestFile(test).hash;
+}
+
+export function is_test_empty(test: string) {
+    return raw.fromTestFile(test).ranges.length === 0;
+}
+
+export function is_edit_from_same_generated(test_edit: string, generated: string) {
+    return raw.fromEditFile(test_edit).content.replace(/\[\[\[|\]\]\]/g, '') === generated;
+}
+
+export function is_edit_empty(test_edit: string) {
+    return !/\[\[\[[^\]]*\]\]\]/.test(raw.fromEditFile(test_edit).content);
+}
+
+export function handler(original_text: string, generated_text: string, mappings: Mappings) {
```

Comment:
please add a short jsdoc class description and also possibly find a more descriptive method name